pkg/storage/v1/storage_nocache.go

```diff
@@ -112,11 +116,48 @@ func (sc StorageNoCache) CreateApplicationProfile(profile *v1beta1.ApplicationPr
 	return nil
 }
 
-func (sc StorageNoCache) PatchApplicationProfile(name, namespace string, patch []byte) error {
-	_, err := sc.StorageClient.ApplicationProfiles(namespace).Patch(context.Background(), name, types.JSONPatchType, patch, metav1.PatchOptions{})
+func (sc StorageNoCache) PatchApplicationProfile(name, namespace string, patch []byte, channel chan error) error {
+	profile, err := sc.StorageClient.ApplicationProfiles(namespace).Patch(context.Background(), name, types.JSONPatchType, patch, metav1.PatchOptions{})
 	if err != nil {
 		return fmt.Errorf("patch application profile: %w", err)
 	}
+	// check if returned profile is full
+	if s, ok := profile.Annotations[iidhelpers.StatusMetadataKey]; ok {
+		if s == iidhelpers.TooLarge {
+			if channel != nil {
+				channel <- utils.FullApplicationProfileError
+			}
+		}
+		return nil
+	}
+	// check if returned profile is too big
+	if s, ok := profile.Annotations[iidhelpers.ResourceSizeMetadataKey]; ok {
+		size, err := strconv.Atoi(s)
+		if err != nil {
+			return fmt.Errorf("parse size: %w", err)
+		}
+		if size > MaxApplicationProfileSize {
+			// add annotation to indicate that the profile is full
```

Comment:
@dwertent please note the application profile won't be truncated to MaxApplicationProfileSize